### DISCUSSION TITLE
Have you considered using Zig instead of Rust?

### DISCUSSION THREAD

**Discussion by NahtanN** [tursodatabase/turso#571] [Q&A] [answered]
Zig is still a young language with a lot of room to grow, but it’s cool to see projects like TigerBeetle being built with it. Did you ever think about using Zig? If so, what made you decide one way or the other?

**Comment by penberg** [comment, marked as answer]
First version of Limbo was actually written in Zig. The plan was to reuse as much of the SQLite front-end as possible and only reimplement the VM with support for asynchronous I/O. However, as I made more progress, I gave up on the idea of reusing SQLite code (because it's pretty complex). At the same time, I also noticed how I was spending much more time in `gdb` and `valgrind` than usual, so I just switched to what I know well, and that's Rust.

**Reply by wearypossum4770** [reply]
Thats funny I was playing with zig a bit. But I have an idea, have you considered including "enum" support to support SQL scripts that have those in it. Maybe it would forced the SQLite Core team to think about including it in their implementation of SQLite4.

**Reply by penberg** [reply]
The focus right now is on SQLite compatibility, not really extending the feature set that much. However, there are some extensions in libSQL around schema management that we definitely want in. The big one that I want to support (possibly before SQLite) is `BEGIN CONCURRENT` but that's for much later.
